clusterscope/cluster_info.py

```diff
@@ -86,12 +132,9 @@ def get_cpu_count(self, timeout: int = 60) -> int:
             RuntimeError: If unable to retrieve CPU information.
         """
         try:
-            result = subprocess.check_output(
-                ["sysctl", "-n", "hw.ncpu"], text=True, timeout=timeout
-            )
-
+            result = run_cli(["sysctl", "-n", "hw.ncpu"], text=True, timeout=timeout)
             return int(result.strip())
-        except (subprocess.SubprocessError, FileNotFoundError) as e:
+        except RuntimeError as e:
             raise RuntimeError(f"Failed to get CPU information: {str(e)}")
```

Comment:
maybe just let it raise on run_cli?

```suggestion
result = run_cli(["sysctl", "-n", "hw.ncpu"], text=True, timeout=timeout)
```

right now we're raising on `run_cli` then grabbing the exception here and raising it again. _seems_ better to just raise it once in `run_cli`?